arcade/scene.py

```diff
@@ -113,8 +110,11 @@ def add_sprite_list(
         """
         if sprite_list is None:
             sprite_list = SpriteList(use_spatial_hash=use_spatial_hash)
-        self.name_mapping[name] = sprite_list
-        self.sprite_lists.append(sprite_list)
+        if name in self._name_mapping.keys():
+            self.remove_sprite_list_by_name(name)
+            warn("A Spritelist with the name: "+name+", is already in the scene, will override Spritelist")
```

Comment:
It's usually better to have a hard error than doing a warning. I'm not sure if this should be a warning/error. Might need some more input from user users/devs. @Cleptomania have any opinion on this?

Reply:
There might be a case where the user doesn't care.